frontend/src/pages/settings/ExecutorsSettings.tsx

```diff
@@ -0,0 +1,666 @@
+    const updatedProfiles = {
+      ...localParsedProfiles,
+      executors: {
+        ...localParsedProfiles.executors,
+        [selectedExecutorType]: {
+          ...localParsedProfiles.executors[selectedExecutorType],
+          [selectedConfiguration]: {
+            [selectedExecutorType]: formData,
+          },
+        },
+      },
+    };
+
+    // Update state
+    setLocalParsedProfiles(updatedProfiles);
+
+    // Save the updated profiles directly
+    try {
+      const contentToSave = JSON.stringify(updatedProfiles, null, 2);
+
+      await saveProfiles(contentToSave);
+      setProfilesSuccess(true);
+      setIsDirty(false);
+      setTimeout(() => setProfilesSuccess(false), 3000);
+
+      // Update the local content as well
+      setLocalProfilesContent(contentToSave);
+    } catch (err: any) {
+      console.error('Failed to save profiles:', err);
+    }
+  };
+
+  if (profilesLoading) {
+    return (
+      <div className="flex items-center justify-center py-8">
+        <Loader2 className="h-8 w-8 animate-spin" />
+        <span className="ml-2">Loading executor configurations...</span>
+      </div>
+    );
+  }
+
+  return (
+    <div className="space-y-6">
+      {!!profilesError && (
+        <Alert variant="destructive">
+          <AlertDescription>
+            {profilesError instanceof Error
+              ? profilesError.message
+              : String(profilesError)}
+          </AlertDescription>
+        </Alert>
+      )}
+
+      {profilesSuccess && (
+        <Alert className="border-green-200 bg-green-50 text-green-800 dark:border-green-800 dark:bg-green-950 dark:text-green-200">
+          <AlertDescription className="font-medium">
+            ✓ Executor configurations saved successfully!
+          </AlertDescription>
+        </Alert>
+      )}
+
+      <Card>
+        <CardHeader>
+          <CardTitle>Coding Agent Configurations</CardTitle>
+          <CardDescription>
+            Customize the behavior of coding agents with different executor
+            profiles.
+          </CardDescription>
+        </CardHeader>
+        <CardContent className="space-y-4">
+          {/* Editor type toggle */}
+          <div className="flex items-center space-x-2">
+            <Checkbox
+              id="use-form-editor"
+              checked={useFormEditor}
+              onCheckedChange={(checked) => setUseFormEditor(!!checked)}
+              disabled={profilesLoading || !localParsedProfiles}
+            />
+            <Label htmlFor="use-form-editor">Edit visually</Label>
+          </div>
+
+          {useFormEditor &&
+          localParsedProfiles &&
+          localParsedProfiles.executors ? (
+            // Form-based editor
+            <div className="space-y-4">
+              <div className="grid grid-cols-2 gap-4">
+                <div className="space-y-2">
+                  <Label htmlFor="executor-type">Executor</Label>
+                  <Select
```

Comment:
`DEFAULT` should always be the first item in the drop down. `ROUTER` is currently the first item for `CLAUDE_CODE`.